src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FindTypeMapObjectsStep.cs

```diff
@@ -0,0 +1,87 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+using Mono.Linker.Steps;
+using Xamarin.Android.Tasks;
+
+namespace MonoDroid.Tuner;
+
+/// <summary>
+/// Scans an assembly for JLOs that need to be in the typemap and writes them to an XML file.
+/// </summary>
+public class FindTypeMapObjectsStep : BaseStep, IAssemblyModifierPipelineStep
+{
+	public bool Debug { get; set; }
+
+	public bool ErrorOnCustomJavaObject { get; set; }
+
+	public TaskLoggingHelper Log { get; set; }
+
+	public FindTypeMapObjectsStep (TaskLoggingHelper log) => Log = log;
```

Comment:
As a "ha ha only serious" / "future coding guidelines" question/discussion: should we start using [C# primary constructors](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-12.0/primary-constructors)?

```csharp
public partial class FindTypeMapObjectStep (TaskLoggingHelper Log) : BaseStep, IAssemblyModifierPipelineStep {
    // …
}
```

Reply:
I don't love primary constructors, but if that's what we decide we want to standardize as a team it won't kill me.  😁

Reply:
I'm not a fan of primary constructors either... :)